content/specific-skill-success-criteria/classes-and-objects/_index.md

```diff
@@ -23,3 +23,118 @@ All students need to provably understand all of the following concepts:
 - getters and setters
 - inheritance and overriding and extending methods
 - composition
+
+# Python
+
+## Constructors
+The `__init__()` method is called the constructor and is always called when an object is created, generally used for instantiating an object.
+
+### Default Constructor
+A simple constructor which doesn’t accept any arguments. Its definition has only one argument which is a reference to the instance being constructed.
+```
+class BankAccount:
+  def __init__(self):
+    pass
+```
+
+### Parameterized Constructor
+A constructor with parameters. Takes its first argument as a reference to the instance being constructed known as self and the rest of the arguments are provided by the programmer.
+```
+class BankAccount:
+  def __init__(self, acc_no, amount):
+    self.acc_no = acc_no
+    self.amount = amount
+```
+
+## Instantiation
+Instantiating a class is creating a copy of the class which inherits all class variables and methods. We simply call the class as if it were a function, passing the arguments that the `__init__()` method defines (except `self`).
+```
+saitama_acc = BankAccount("sa1234", 10000)
+```
+
+A class can have multiple instances.
+```
+kuroko_acc = BankAccount("ka1234", 300)
+tsukasa_acc = BankAccount("ta1234", 75010)
+vash_acc = BankAccount("va1234", 813650)
+```
+
+
+## Getters and setters
+
+### Getters
+A getter is a method that gets the value of a property.
+```
+  def balance(self):
+    return self.amount
+```
+
+```
+saitama_acc.amount
+```
+
+```
+saitama_acc.balance()
+```
+
+### Setters
+A setter is a method that sets the value of a property.
+```
+  def deposit(self, amount):
+    self.amount += amount
+
+  def withdraw(self, amount):
+    self.amount -= amount
+```
+
+```
+saitama_acc.deposit(500)
+saitama_acc.withdraw(320)
+```
+
+
+## Inheritance
+It enables us to create a new class from an existing class. The new class (child class) is a specialized version of the existing class (parent class) and it inherits all the non-private variables and methods of the existing class.
```

Comment:
I'd say:

Inheritance enables us to create a new class from an existing class. The new class (child class) is a specialized version of the existing class (parent class) and it inherits all the non-private variables and methods of the existing class.